stackrox/stackrox
# ROX-17275: Replace classic Cluster Health widget with PatternFly cards in System Health

## Description

TL;DR

Replace classic Cluster Health widget which had 6 sub-widgets with 3 PatternFly cards:
* Cluster status
* Sensor upgrade
* Credential expiration

### Problems

1. Cluster overview omits Unavailable status.
3. Cause and effect for status is unclear.
3. Cards display only most severe status.
4. Color vision accessibility.
5. Tight coupling with clusters for colors and icons (my bad).

### Analysis

Reoganize information following example of declarative configuration to distinguish no errors versus errors:
* Card has head with green check mark, but not body, if non-zero healthy and zero unhealthy and degraded.
* Card has body for request error or for table of counts if zero healthy or any non-zero unhealthy or degraded (which have red or yellow icons).
    * 4 rows make cause and effect relationship clearer in cluster status table.
    * 5 status and total columns make relationship clearer in all tables.

### Solution

1. Move clusters request from DashboardPage to ClustersHealthCards.
2. Begin to separate files in Components to more specific subfolders like ClustersHealth.
3. Delete obsolete components.
4. Rewrite integration tests.

## Checklist

## Testing Performed

### Manual testing

1. Visit classic /main/system-health: see height of Cluster Health widget

    * at 1440px screen width
        ![classic_1440](https://github.com/stackrox/stackrox/assets/11862657/52f891f9-1614-4e46-8067-ad3e3285172b)

    * at 1920px screen width
        ![classic_1920](https://github.com/stackrox/stackrox/assets/11862657/0daf319f-e240-4d28-9a58-b38d32208129)

2. Visit PatternFly /main/system-health

    * spinner until response from **first** GET /v1/clusters request
        ![spinner](https://github.com/stackrox/stackrox/assets/11862657/1194ff64-9315-42c2-8d24-b372fae44e69)

    * request error message if I temporarily mangle the endpoint address (similarly, Network Error when infra cluster stops)
        ![requestErrorMessage](https://github.com/stackrox/stackrox/assets/11862657/5ce78c83-8884-4f19-a35b-1f6ce09d99f7)

    * card header only if **healthy**
        ![healthy](https://github.com/stackrox/stackrox/assets/11862657/1b4a47c7-850a-43bb-8b03-48e5a5c4d964)

### Integration testing

1. `yarn cypress-open` in ui

    * systemHealth/clusters.test.js

    * fixture with 7 clusters: see red icons for non-zero counts in unhealthy columns and yellow icons for non-zero counts in degraded columns
      ![fixture_7_Th](https://github.com/stackrox/stackrox/assets/11862657/ad3acfba-e5c0-4531-bd31-ad8f6ef11312)

    * fixture with 3 clusters: see Sensor upgrade is healthy, therefore does not need card body with counts table
      ![fixture_3_Th](https://github.com/stackrox/stackrox/assets/11862657/4960eee1-85ea-49af-aa42-24a12ddd7c9e)

    * fixture with 1 cluster that has Uninitialized status: see counts tables because no healthy clusters
        ![fixture_1_Uninitialized_Th](https://github.com/stackrox/stackrox/assets/11862657/45de8923-dc90-4ec2-927a-a468b481b016)
